packages/bento-design-system/src/BentoConfigContext.tsx

```diff
@@ -0,0 +1,25 @@
+import { createContext, useContext } from "react";
+import * as defaultConfigs from "./util/defaultConfigs";
+import { BentoConfig, PartialBentoConfig } from "./BentoConfig";
+import merge from "ts-deepmerge";
+import { Children } from "./util/Children";
+
+const BentoConfigContext = createContext<BentoConfig>(defaultConfigs);
+
+export function useBentoConfig() {
+  return useContext(BentoConfigContext);
+}
+
+export function BentoConfigProvider({
+  value: config,
+  children,
+}: {
+  value: PartialBentoConfig;
+  children: Children;
+}) {
+  return (
+    <BentoConfigContext.Provider value={merge(defaultConfigs, config)}>
+      {children}
+    </BentoConfigContext.Provider>
+  );
+}
```

Comment:
This is where we merge default and user config, and propagate it down via context

Reply:
ain't we solving https://github.com/buildo/bento-design-system/issues/277 here, right?

Reply:
I don't expect this change to have any effect on it, no.